src/Bicep.Core/TypeSystem/CyclicCheckVisitor.cs

```diff
@@ -149,15 +155,47 @@ public override void VisitFunctionCallSyntax(FunctionCallSyntax syntax)
             base.VisitFunctionCallSyntax(syntax);
         }
 
-        public override void VisitObjectTypePropertySyntax(ObjectTypePropertySyntax syntax)
+        public override void VisitNullableTypeSyntax(NullableTypeSyntax syntax)
         {
```

Comment:
As discussed offline, might be worth exploring an approach which utilizes the visitor state itself rather than an ad-hoc check using the syntax hierarchy.

Reply:
I couldn't find a reasonable way to implement this check in `CyclicCheckVisitor` without reimplementing some of the model's type manager, but the check can be made much simpler if it's performed after type checking has been performed. I added a second cyclic checker that runs as part of EmitLimitationCalculator.